### PR TITLE
split getting started guide into a view and a form component

### PR DESCRIPTION
**Pull request by salsakran**
This is another in the series of cleaning up the reference guide. This is a precursor to fixing #5106.

Would love a sanity check as well as advice on how we usually deal with forms. Should I keep the `isEditing` flag and use the same route, or have a separate `reference/edit` route and a different container for edit state. IMO the latter has advantages in that the data fetching can be split off as opposed to switched on isEditing, but I don't have strong feelings about this. 

Do any of you @tlrobinson @attekei @camsaul @kdoh  ?

### CODE REVIEW

**Comment by kdoh** [comment]
I think my preference is to have the separate route and separate component. I think that'll end up being a bit clearer and make each respective component more focused and easier to work with.

File: frontend/src/metabase/reference/guide/GettingStartedGuide.jsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                        />
+
+                        <div className="wrapper wrapper--trim">
+                            { (!guide || isGuideEmpty(guide)) && user && user.is_superuser && (
+                                <AdminInstructions>
+                                    <h2 className="py2">Help your team get started with your data.</h2>
+                                    <GuideText>
+                                        Show your team what’s most important by choosing your top dashboard, metrics, and segments.
+                                    </GuideText>
+                                    <button
```
Comment (salsakran) [inline frontend/src/metabase/reference/guide/GettingStartedGuide.jsx]: @kdoh if I split up the routes, then this button should take you to `/reference/edit`. What's the usual way of doing this? An <a> or changing the startEditing function to go to that url?

Comment (kdoh) [inline frontend/src/metabase/reference/guide/GettingStartedGuide.jsx]: Seems like you should use a `<Link>` element with the `to` prop pointing to that route. You can then add various `Button` classes to have it show up as a button.

**Comment by salsakran** [comment]
Ok, so I'd like to merge this and do the container split when @cam and I pow-wow about a more performant way to update the api endpoint for this. (Short form: everything is normalized, the guide endpoint only returns IDs, so effectively the client has to have *all* the metadata about *everything* clientside.)

**Review by attekei — APPROVED** [review]
LGTM and seems that everything works.

Would be helpful to have some commentary along what for instance `exploreLinksForMetric` is about but that isn't absolutely necessary.

(Agreed re: separation of form edit state to a separate component)

**Comment by salsakran** [comment]
@attekei added some comments about that function. I agree  that "expore links" is somewhat vague.

**Pull request merged**